zipline/data/hdf5_daily_bars.py

```diff
@@ -240,75 +267,54 @@ def write(self, country_code, frames, scaling_factors=None):
         if scaling_factors is None:
             scaling_factors = DEFAULT_SCALING_FACTORS
 
-        with self.h5_file(mode='a') as h5_file:
```

Comment:
This block was getting long enough that I broke it out into subroutines that write each subgroup.